plugins/mirror/bazel.go

```diff
@@ -40,6 +40,35 @@ func (a *Artifact) AppendURL(url string) {
 	a.rule.Attr("urls").(*build.ListExpr).List = append(list, &build.StringExpr{Value: url})
 }
 
+type HTTPClient interface {
+	Head(uri string) (resp *http.Response, err error)
+	Get(uri string)  (resp *http.Response, err error)
+}
+
+var Client HTTPClient
+
+func init() {
+	Client = http.DefaultClient
```

Comment:
As soon as something like this is done, one knows, it is the time for structs.

In a follow-up :)

Reply:
Can you explain what you mean by that?

Reply:
I mean that replacing dependencies globally in tests has the tendency to lead to a situation where one mocks something for test A, forgets that things are mocked, does not revert it and test B fails in strange ways.

A proper struct, can encapsulate the http client. It is side-effect free.

Reply:
I understand the problem, and I agree that replacing a global variable can impose the risk of forgetting to reset to the original value. I would have solved this by directly passing the httpClient as an argument instead. Can you point me to an example where it is done like you explain?

Reply:
Argument is fine too.